code/modules/mining/mine_structures.dm

```diff
@@ -40,7 +40,8 @@
 		"<span class='notice'>You put \a [W.name] on the \the [src].</span>")
 		playsound(get_turf(src), 'sound/machines/click.ogg', 20, 1)
 		qdel(W)
-		new /obj/structure/hanging_lantern(get_turf(src), src.dir)
+		var/obj/structure/hanging_lantern/L = new/obj/structure/hanging_lantern(get_turf(src)) // Obligatory "fuck dylan".
```

Comment:
by the way, you can just write "var/obj/structure/hanging_lantern/L = new(get_turf(src))"